include/modules/hidelist.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2018 Dylan Frank <b00mx0r@aureus.pw>
+ *
+ * This file is part of InspIRCd.  InspIRCd is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#pragma once
+
+#include "event.h"
+
+class HidelistEventListener : public Events::ModuleEventListener
+{
+ public:
+	HidelistEventListener(Module* mod)
+		: ModuleEventListener(mod, "event/hidelist")
+	{
+	}
+
+	/** Called whenever a /WHOIS is performed by a local user.
+	 * @param whois Whois context, can be used to send numerics
+	 */
+	virtual ModResult OnListDenied(User* user, Channel* chan, const std::string& modename) = 0;
```

Comment:
Pass a mode handler (or maybe a mode reference?) here.

Reply:
We don't have this information in the ModeWatcher class. Also, the mode name just needs to be passed for the m_override snomask announcement.